sdk/inc/azure/core/internal/az_event_pipeline_internal.h

```diff
@@ -0,0 +1,115 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file
+ *
+ * @brief Definition of #_az_event_pipeline and related types describing a bi-directional HFSM
+ *        pipeline.
+ *
+ */
+
+#ifndef _az_EVENT_PIPELINE_INTERNAL_H
+#define _az_EVENT_PIPELINE_INTERNAL_H
+
+#include <azure/core/az_event.h>
+#include <azure/core/az_event_policy.h>
+#include <azure/core/az_platform.h>
+#include <azure/core/az_result.h>
+#include <azure/core/internal/az_hfsm_internal.h>
+#include <stdint.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+// Definition is below.
+
+typedef struct _az_event_pipeline _az_event_pipeline;
+
+/**
+ * @brief Internal definition of a pipeline.
+ */
+struct _az_event_pipeline
+{
+  struct
+  {
+    az_event_policy* outbound_policy;
+    az_event_policy* inbound_policy;
+    az_platform_mutex mutex;
+  } _internal;
+};
+
+/**
+ * @brief Initializes a pipeline.
+ *
+ * @param[out] pipeline The #_az_event_pipeline to initialize.
+ * @param[in] outbound The #az_event_policy at the outbound end of the pipeline.
+ * @param[in] inbound The #az_event_policy at the inbound end of the pipeline.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK Success.
```

Comment:
I think we wanted to have some other retvals added here in case they are possible.
In other _init cases we have a generic " * @return An #az_result value indicating the result of the operation."

Reply:
Looking at the az_json_writer:

```
 * @retval #AZ_OK #az_json_writer is initialized successfully.
 * @retval other Initialization failed.
```